### ISSUE TITLE
Regression in zengularity/benji v2.2.0

### ISSUE THREAD

**Issue by dwijnand** [lampepfl/dotty#16337]
```scala
[error] -- [E008] Not Found Error: /home/jenkins/agent/workspace/buildCommunityProject/repo/s3/src/test/scala/S3Spec.scala:62:9 
[error] 62 |      ls.list(Option.empty[String])({ t =>
[error]    |      ^^^^^^^
[error]    |value list is not a member of bucket.ListRequest - did you mean ls.wait?
[error] -- [E051] Reference Error: 


/home/jenkins/agent/workspace/buildCommunityProject/repo/s3/src/test/scala/S3Spec.scala:30:34 
[error] 30 |    "paginate the bucket objects" in withMatEx { implicit ee: EE =>
[error]    |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[error]    |Ambiguous overload. The overloaded alternatives of method in in class BlockExample with types
[error]    | (execution: org.specs2.specification.core.Execution):
[error]    |  org.specs2.specification.core.Fragment
[error]    | [R]
[error]    |  (parser: org.specs2.specification.script.StepParser[R])
[error]    |    (implicit evidence$6: org.specs2.execute.AsResult[R]):
[error]    |      org.specs2.specification.core.Fragment
[error]    | (fs: => org.specs2.specification.core.Fragments)
[error]    |  (implicit p1: org.specs2.control.ImplicitParameters.ImplicitParam1):
[error]    |    org.specs2.specification.core.Fragments
[error]    | (f: => org.specs2.specification.core.Fragment):
[error]    |  org.specs2.specification.core.Fragment
[error]    | [R]
[error]    |  (f: String => R)
[error]    |    (implicit ar: org.specs2.specification.core.AsExecution[R]):
[error]    |      org.specs2.specification.core.Fragment
[error]    | [R]
[error]    |  (r: => R)
[error]    |    (implicit evidence$4: org.specs2.specification.core.AsExecution[R]):
[error]    |      org.specs2.specification.core.Fragment
[error]    |all match arguments (Nothing)
[error]    |
[error]    | longer explanation available when compiling with `-explain`
[error] two errors found
[error] (s3 / Test / compileIncremental) Compilation failed
```

```scala
 trait ListRequest 
 def test() = {
   type StructType = {
     def list(nextToken: Option[String])(andThen: String => Source[Object, akka.NotUsed])(implicit  m:Materializer): Source[Object, akka.NotUsed]
    }
   val ls: bucket.ListRequest with StructType = ???
   ls.list(Option.empty[String])({ t =>
        tok1 = Some(t)
        Source.empty[Object]
      })
}

**Comment by odersky** [comment]
would be good to check again. I hope it is fixed by #16344.

**Comment by WojciechMazur** [comment]
Project is passing OpenCB with the latest compiler
